services/ingestor/src/main/kotlin/company/thewebhook/ingestor/Application.kt

```diff
@@ -47,13 +50,19 @@ fun Application.module() = launch {
             .lowercase()
             .split(",")
 
+    val serverHostRegexString =
+        environment.config
+            .propertyOrNull("ingestor.regex.serverhost")?.getString()
+            ?: throw ServerHostRegexException("Serverhost regex string not provided")
+
     val producer by inject<Producer<ByteArray>>()
     producer.connect(messageStoreConfig)
 
     routing {
         HttpMethod.DefaultMethods.forEach {
             route("{...}", it) {
                 handle {
+                    val serverHostRegex = Regex(serverHostRegexString)
```

Comment:
Declaring this var on every request will give us performance penalties. Bring it outside `routing{}`.

Also search up compiled regexes - it will give a perf boost.

Reply:
Okay okay

Reply:
I think kotlin.text.regex doesn't have option for compiled regex. Shall we use java.util.regex?